purchase_partial_invoicing/purchase.py

```diff
@@ -19,122 +19,103 @@
 #
 ##############################################################################
 
-from openerp.osv import fields, orm
+from openerp import models, fields, api
 import openerp.addons.decimal_precision as dp
 
 
-class purchase_order_line(orm.Model):
+class purchase_order_line(models.Model):
+
+    @api.multi
+    def _get_invoiced_quantity(self):
+        self.ensure_one()
+        invoiced_qty = 0.0
+        for invoice_line in self.invoice_lines:
+            invoiced_qty += invoice_line.quantity
+        return invoiced_qty
+
+    @api.one
+    @api.depends('invoice_lines', 'invoice_lines.invoice_id',
+                 'invoice_lines.quantity')
+    def compute_invoiced_qty(self):
+        self.invoiced_qty = self._get_invoiced_quantity()
+
+    @api.multi
+    def is_fully_invoiced(self):
+        self.ensure_one()
+        return self.invoiced_qty == self.product_qty
+
+    @api.one
+    @api.depends('invoice_lines', 'invoice_lines.invoice_id',
+                 'invoice_lines.quantity')
+    def compute_fully_invoiced(self):
+        self.fully_invoiced = self.is_fully_invoiced()
+
+    @api.multi
+    def is_all_invoices_approved(self):
+        if self.invoice_lines:
+            return not any(inv_line.invoice_id.state
+                           in ['draft', 'cancel']
+                           for inv_line in self.invoice_lines)
+        else:
+            return False
```

Comment:
I don't think so, because in that case, it will be `any([])` = `False`, and with the previous `not`, it will be evaluated as True.